growattServer/__init__.py

```diff
@@ -580,33 +583,63 @@ def update_plant_settings(self, plant_id, changed_settings, current_settings = N
         data = json.loads(response.content.decode('utf-8'))
         return data
 
-    def update_mix_inverter_setting(self, serial_number, setting_type, parameters):
+
+    def update_inverter_setting(self, serial_number, setting_type, 
+                                default_parameters, parameters, endpoint):
         """
         Applies settings for specified system based on serial number
         See README for known working settings
 
         Keyword arguments:
-        serial_number -- The serial number (device_sn) of the inverter
-        setting_type -- The setting to be configured (list of known working settings below)
-        parameters -- A python dictionary of the parameters to be sent to the system based on the chosen setting_type, OR a python array which will be converted to a params dictionary
+            serial_number (str): Serial number (device_sn) of the inverter
+            setting_type (str): Setting to be configured
+            default_params (dict): Default set of parameters for the setting call
+            parameters (dict or list of str): Parameters to be sent to the system 
+                (array which will be converted to a dictionary)
+            endpoint (str): The URL endpoint that should be called
 
         Returns:
-        A response from the server stating whether the configuration was successful or not
+            JSON response from the server whether the configuration was successful
         """
-        setting_parameters = parameters
-
+        settings_parameters = parameters
+        
         #If we've been passed an array then convert it into a dictionary
         if isinstance(parameters, list):
-            setting_parameters = {}
+            settings_parameters = {}
             for index, param in enumerate(parameters, start=1):
-                setting_parameters['param' + str(index)] = param
+                settings_parameters['param' + str(index)] = param
+        
+        settings_parameters = {**default_parameters, **settings_parameters}
+        print(settings_parameters)
```

Comment:
Don't need a print here

Reply:
Dang, thought I caught everything along the way.

Reply:
And that's why we have reviews :)

I'm at work right now, but my first skim through looks good. I'll take another look this evening and come back to you. Ultimately it's up to @indykoning  though to give final approval & merge